### ISSUE TITLE
Support Image Manifests List Digest

### ISSUE THREAD

**Issue by jeremyruffell** [argoproj-labs/argocd-image-updater#297]
**Is your feature request related to a problem? Please describe.**
I am trying to keep an environment up to date by grabbing the latest _dev_ tag. 
I am currently running a multi arch cluster (arm64 & amd64), this means using the **digest** update strategy is no longer viable as it grabs the amd64 digest e.g. `my-image:dev@amd64-digest-sha`. To get around this I am using the **name** update strategy and tagging my images with `dev-1635719863`  as it sets the image tag to `dev-1635719863` rather than a digest, therefore the container can grab the correct image based off its architecture. Now that I have multiple environments (dev, qa, sandbox, staging & prod) my tag lists are getting messy. 

**Describe the solution you'd like**
Support referencing the **Image Manifest List Digest**, This way I could potentially use the image manifest digest for a tag and not have to do lexical sorting over an environment specific tag e.g. `my-image@dev:image-manifest-digest-sha`

**Describe alternatives you've considered**
I am using the **name** update strategy and tagging my images with `dev-1635719863`  as it sets the image tag to `dev-1635719863` rather than a digest, therefore the container can grab the correct image based off its architecture.

**Additional context**
![image (1)](https://user-images.githubusercontent.com/36174931/140844572-6f94be28-13c2-461b-b884-cc22f477ed3d.jpg)

**Comment by jannfis** [comment]
I agree that we need multi-arch images support for the digest strategy. I haven't dived too deep into multi-arch images, as I'm bound to amd64 platform currently.

I'm happy to read up about it and to discuss the requirements for this.

**Comment by diranged** [comment]
@jannfis Just adding in my support for this issue... we got al the way to being able to ship multi-arch images, and then ran into this with the image-updater:

```
time="2021-12-23T18:38:59Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-d858c49 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:12Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-eef31b3 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:12Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-f42f381 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:12Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-fake-upgrade_action - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:13Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-053b732 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:13Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-3631df4 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:13Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-9edd8cb - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:14Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-e119a59 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:14Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-1 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:14Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:test-b9ba422 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:15Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:latest - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:15Z" level=error msg="Error fetching metadata for dev/hello-world-k8s.template:main-ea37284 - neither V1 or V2 or OCI manifest returned by registry: missing signature key"
time="2021-12-23T18:39:17Z" level=debug msg="found 0 from 54 tags eligible for consideration" image="xxx.dkr.ecr.us-west-2.amazonaws.com/dev/hello-world-k8s.template:main-43e1053"
```

**Comment by dixneuf19** [comment]
Hi, I recently had the same issue on my Raspberry Pie cluster, using ghcr.io as registry, and managed to patch my way out of this. Basically, I added ManifestList as a valid manifest type expected, and if this type of manifest is return, I try to fetch the manifest matching the current arch and os of the runtime. Not perfect since you could have mixed architecture workloads but works for my use case.

See this commit for the diff: https://github.com/dixneuf19/argocd-image-updater/commit/271ca47559fb0ceebfe4ed5ef89affaa8e93e772

I'll try to submit a pull request when I'll have a bit of time.

**Comment by jannfis** [comment]
Thanks @dixneuf19 - I have taken inspiration from your commit, and put in a little more features and fine tuning. Just merged #341

The functionality is not yet fully complete, and lacks documentation (to be written), but it does also support multi-arch clusters as requested by OP.

**Comment by jannfis** [comment]
Documentation supplied with #398
